src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -277,7 +276,11 @@ public bool IsBetween(TimeOnly start, TimeOnly end)
         /// <param name="t1">The first TimeOnly instance.</param>
         /// <param name="t2">The second TimeOnly instance..</param>
         /// <returns>The elapsed time between t1 and t2.</returns>
-        public static TimeSpan operator -(TimeOnly t1, TimeOnly t2) => new TimeSpan((t1._ticks - t2._ticks + TimeSpan.TicksPerDay) % TimeSpan.TicksPerDay);
+        public static TimeSpan operator -(TimeOnly t1, TimeOnly t2)
+        {
+            long diff = (long)(t1._ticks - t2._ticks);
+            return new TimeSpan(diff + ((diff >> 63) & TimeSpan.TicksPerDay));
```

Comment:
> return new TimeSpan(diff + ((diff >> 63) & TimeSpan.TicksPerDay)); [](http://example.com/codeflow?start=11&length=67)

will be good to add a comment telling what this operation does. It is not obvious what it does anymore.